File: abc/abc147/abc147_b.cs
// https://atcoder.jp/contests/abc147/submissions/8850430

using System;

namespace CSharp
{
    class Program
    {
        static void Main()
        {
            var S = Console.ReadLine();

            var result = 0;
            for (int i = 0; i < S.Length / 2; i++)
            {
                if (S[i] != S[S.Length - 1 - i])
                {
                    result++;
                }
            }
            Console.WriteLine(result);
        }
    }
}

File: abc/abc147/abc147_b.fs
// https://atcoder.jp/contests/abc147/submissions/12160720

let S = stdin.ReadLine().ToCharArray()

[| 0..S.Length / 2 - 1 |]
|> Array.filter (fun i -> S.[i] <> S.[S.Length - 1 - i])
|> Array.length
|> stdout.WriteLine
